### PR TITLE
Sam/fix create eos

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

- fixed: Crash in `createPrivateKey` for `EosTools`

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204323976548769

### CODE REVIEW

File: src/eos/eosPlugin.ts [hunk 1]
```diff
@@ -79,10 +81,16 @@ export class EosTools implements EdgeCurrencyTools {
 
     const currencyInfoType = this.currencyInfo.walletType.replace('wallet:', '')
     if (type === currencyInfoType) {
-      let entropy = Buffer.from(this.io.random(32)).toString('hex')
-      const eosOwnerKey = PrivateKey.from(entropy).toWif()
-      entropy = Buffer.from(this.io.random(32)).toString('hex')
-      const eosKey = PrivateKey.from(entropy).toWif()
+      const ownerKeyEntropy = Buffer.from(this.io.random(32)).toString('hex')
```
Comment (swansontec) [inline src/eos/eosPlugin.ts:84]: The bytes / buffer / hex conversion is a bit weird here. Can their `Bytes` class just take a Uint8Array, as returned by the `random` function? If we really need to go hex, use `base16` from rfc4648. Maybe also put the whole random -> Bytes thing on one line, so we aren't converting halfway on one line, and then converting more on the next.

File: src/eos/eosEngine.ts [hunk 2]
```diff
@@ -198,15 +198,6 @@ export class EosEngine extends CurrencyEngine<EosTools, SafeEosWalletInfo> {
...
     await super.loadEngine(plugin, walletInfo, opts)
-
-    if (typeof this.walletInfo.keys.ownerPublicKey !== 'string') {
-      if (walletInfo.keys.ownerPublicKey != null) {
-        this.walletInfo.keys.ownerPublicKey = walletInfo.keys.ownerPublicKey
-      } else {
-        const pubKeys = await plugin.derivePublicKey(this.walletInfo)
-        this.walletInfo.keys.ownerPublicKey = pubKeys.ownerPublicKey
-      }
-    }
```
Comment (swansontec) [inline src/eos/eosEngine.ts:209]: Ahhh, this was before the days where the core did the public key derivation for us. Yes, remove it.

**Review by swansontec — APPROVED** [review]

File: src/eos/eosPlugin.ts [hunk 3]
```diff
@@ -98,24 +98,22 @@ export class EosTools implements EdgeCurrencyTools {
...
-      if (walletInfo.keys.eosOwnerKey != null) {
-        ownerPublicKey = PrivateKey.from(walletInfo.keys.eosOwnerKey)
-          .toPublic()
-          .toLegacyString()
-      }
-      return { publicKey, ownerPublicKey }
-    } else {
-      throw new Error('InvalidWalletType')
     }
+    return { publicKey, ownerPublicKey }
```
Comment (swansontec) [inline src/eos/eosPlugin.ts:116]: Hide whitespace to view this one. It's another cleanup we should do while we were in there - I wanted to be sure your PR made sense, but this guy was all weird.

File: src/eos/eosEngine.ts (hunk 2 above)
Comment (samholmes) [reply src/eos/eosEngine.ts:209]: This code predates `derivePublicKey`?

Comment (swansontec) [reply src/eos/eosEngine.ts:209]: I believe so, yes. Early on, engines would just get the private keys, and were on their own in terms of deriving public keys. This was slow, and everybody did it differently, so we moved it to the core. A few old coins still do it themselves, redundantly.

**Pull request merged**